infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -597,6 +634,8 @@ protected void startMachines() throws InfrastructureException {
    */
   protected void doStartMachine(KubernetesServerResolver serverResolver)
       throws InfrastructureException {
+    machineStartupTraces = new ConcurrentHashMap<>(getContext().getEnvironment().getPods().size());
```

Comment:
`#` of machines `!=` `#` of pods. Am I misunderstanding or this value should be predicted size for a map for a # of machines?

Reply:
That is true. To compute the precise size, one would have to go through all the specs of the pods and count the number of containers. I opted to just "estimate" the initial size by assuming that, on average, a single pod has a single container. That assumption might be wrong though. WDYT?

Reply:
It is our goal to have pod == machine, but for now, it is not. In particular, in the current state of Che 7 we have just 1 pod for all the machines.

Reply:
Would `getContext().getEnvironment().getMachines().size()` work here?

Reply:
It would! Thanks for the suggestion.